presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2747,18 +2747,23 @@ private OptionalLong analyzeParameterAsRowCount(Parameter parameter, Scope scope
                 // validate parameter index
                 analyzeExpression(parameter, scope);
                 Expression providedValue = analysis.getParameters().get(NodeRef.of(parameter));
+                Object value;
                 try {
-                    return OptionalLong.of((long) ExpressionInterpreter.evaluateConstantExpression(
+                    value = ExpressionInterpreter.evaluateConstantExpression(
                             providedValue,
                             BIGINT,
                             metadata,
                             session,
                             accessControl,
-                            analysis.getParameters()));
+                            analysis.getParameters());
                 }
                 catch (VerifyException e) {
                     throw semanticException(INVALID_ARGUMENTS, parameter, "Non constant parameter value for LIMIT: %s", providedValue);
                 }
+                if (value == null) {
+                    throw semanticException(INVALID_ARGUMENTS, parameter, "Parameter value provided for %s is NULL: %s", nonConstantErrorContext, providedValue);
```

Comment:
`nonConstantErrorContext` isn't defined anywhere in this commit or prior to it, so this will produce a compile-time error.

Reply:
I messed the order of commits during rebase.